### PR TITLE
one step checkout support / disable bolt on payment only checkout pages

### PR DESCRIPTION
**Pull request by danac-gs**

### CODE REVIEW

File: etc/adminhtml/system.xml [hunk 1]
```diff
@@ -141,6 +141,10 @@
                     <label>Use Store Credit on Shopping Cart</label>
                     <source_model>Magento\Config\Model\Config\Source\Yesno</source_model>
                 </field>
+                <field id="enable_payment_only_checkout" translate="label" type="select" sortOrder="250" showInDefault="1" showInWebsite="1" showInStore="1">
```
Comment (daisy1754) [inline etc/adminhtml/system.xml]: shouldn't this be inferred via presence of paymentonly publishable key?

File: Block/Checkout/ComponentSwitcherProcessor.php [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    /**
+     * Process the layout
+     *
+     * @param array $jsLayout
+     * @return array
+     */
+    public function process($jsLayout) {
+        // Payment only checkout based on Magento default one-page checkpout
+        // (one-page checkout, Mageplaza one-step checkout etc.)
+        $jsLayout['components']['checkout']['children']['steps']['children']['billing-step']['children']['payment']
```
Comment (daisy1754) [inline Block/Checkout/ComponentSwitcherProcessor.php]: is it possible any of it is undefined, like `$jsLayout['components']['checkout']['children']['steps']` is undefined?

File: etc/adminhtml/system.xml (hunk 1 above)
Comment (danac-gs) [reply etc/adminhtml/system.xml]: @daisy1754 Yes but the solution was front-end based and had an error as well so we had issues with checkout that is not the default Magento one-page checkout but 3rd party (Magewrox one-step checkout on Coastal Business). This way it is removed from layout completely.

File: Block/Checkout/ComponentSwitcherProcessor.php (hunk 2 above)
Comment (danac-gs) [reply Block/Checkout/ComponentSwitcherProcessor.php]: No, it is defined in our layout extension 
https://github.com/BoltApp/bolt-magento2/blob/one-step-checkout-support/view/frontend/layout/checkout_index_index.xml

**Pull request merged**